### PR TITLE
Misc fixes

### PR DESCRIPTION
**Pull request by anba**

### CODE REVIEW

File: spec.html [hunk 1]
```diff
@@ -178,7 +178,10 @@
     <emu-grammar>UnaryExpression : `delete` UnaryExpression</emu-grammar>
     <ul>
       <li>
-        It is a Syntax Error if the |UnaryExpression| is contained in strict mode code and the derived |UnaryExpression| is <emu-grammar>PrimaryExpression : IdentifierReference</emu-grammar>, <ins><emu-grammar>MemberExpression : MemberExpression `.` PrivateName</emu-grammar>, or <emu-grammar>CallExpression : CallExpression `.` PrivateName</emu-grammar></ins>.
+        It is a Syntax Error if the |UnaryExpression| is contained in strict mode code and the derived |UnaryExpression| is <emu-grammar>PrimaryExpression : IdentifierReference</emu-grammar>.
+      </li>
+      <li>
+        <ins>It is a Syntax Error if the derived |UnaryExpression| is <emu-grammar>MemberExpression : MemberExpression `.` PrivateName</emu-grammar> or <emu-grammar>CallExpression : CallExpression `.` PrivateName</emu-grammar></ins>.
```
Comment (littledan) [inline spec.html]: We discussed this previously in https://github.com/tc39/proposal-class-fields/issues/30 . As described there, I don't think this change is necessary.

File: spec.html [hunk 2]
```diff
@@ -417,6 +417,7 @@
       1. Perform ? DefinePropertyOrThrow(_receiver_, _fieldName_, _desc_).
     <emu-alg>
 </emu-clause>
+<emu-note type=editor>TODO: Set the running execution context's PrivateNameEnvironment when calling a function.</emu-note>
```
Comment (littledan) [inline spec.html]: Oops, this is a pretty serious omission! Great catch.

File: spec.html [hunk 3]
```diff
@@ -658,6 +658,7 @@
         1. Let _constructorParent_ be the intrinsic object %FunctionPrototype%.
       1. Else,
         1. Set the running execution context's LexicalEnvironment to _classScope_.
+        1. <ins>NOTE: The running execution context's PrivateNameEnvironment is _outerPrivateEnvironment_ when evaluating |ClassHeritage|.</ins>
```
Comment (littledan) [inline spec.html]: Why is this necessary? Should we have more notes explaining the state of various environments, or is this enough?

File: spec.html [hunk 4]
```diff
@@ -703,7 +703,7 @@
           1. Set the running execution context's LexicalEnvironment to _lex_.
           1. <ins>Set the running execution context's PrivateNameEnvironment to _outerPrivateEnvironment_.</ins>
           1. Return Completion(_status_).
-        1. <ins>Append _fields_ to _fieldRecords_</ins>
+        1. <ins>Append to _fieldRecords_ the elements of _fields_.</ins>
```
Comment (littledan) [inline spec.html]: The English here seems awkward to me (I'd prefer "Append the elements of _fields_ to _fieldRecords_"), but it seems to match usage in other parts of ecma262, so I guess it's appropriate.

**Comment by littledan** [comment]
Thanks for noticing and fixing all of these issues. These changes mostly all look good, though I think ece4643 is unnecessary (but it doesn't hurt).

File: spec.html (hunk 3 above)
Comment (anba) [reply spec.html]: The note in https://tc39.github.io/proposal-class-fields/#sec-execution-contexts gives the impression that implementers could use a single environment for `classScope` and `classPrivateEnvironment`, but during evaluation of `ClassHeritage` the lexical environment is `classScope`, but the private name environment is still `outerPrivateEnvironment`. Which means it's actually not possible to use a single environment for `classScope` and `classPrivateEnvironment`. I think it's worth pointing this out here, isn't it?

File: spec.html (hunk 1 above)
Comment (anba) [reply spec.html]: Ah, I see.

File: spec.html (hunk 3 above)
Comment (littledan) [reply spec.html]: Do you think we should evaluate the heritage in the private name environment created in the class, then?

Comment (anba) [reply spec.html]: Not necessarily. I'd probably leave it as is, and only if there's pushback from implementers we could consider pushing the private name environment before evaluating the class heritage expression.

Comment (littledan) [reply spec.html]: cc @gsathya @xanplz

**Comment by littledan** [comment]
Committed these, omitting the 'delete' change'.
